tests/components/octoprint/test_config_flow.py

```diff
@@ -0,0 +1,331 @@
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_printer_info",
+        return_value=OctoprintPrinterInfo(
+            {
+                "state": {
+                    "flags": {"printing": True, "error": False},
+                    "text": "Operational",
+                },
+                "temperature": [],
+            }
+        ),
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_job_info",
+        return_value=OctoprintJobInfo(
+            {
+                "job": {},
+                "progress": {"completion": 50},
+            }
+        ),
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.request_app_key", return_value="test-key"
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"username": "testuser"}
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+
+
+async def test_import_yaml(hass: HomeAssistant) -> None:
+    """Test that the yaml import works."""
+    with patch(
+        "pyoctoprintapi.OctoprintClient.get_server_info",
+        return_value=True,
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_discovery_info",
+        return_value=DiscoverySettings({"upnpUuid": "uuid"}),
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.request_app_key", return_value="test-key"
+    ), patch(
+        "homeassistant.components.octoprint.async_setup", return_value=True
+    ), patch(
+        "homeassistant.components.octoprint.async_setup_entry",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                "host": "1.1.1.1",
+                "api_key": "test-key",
+                "name": "Printer",
+                "port": 81,
+                "ssl": True,
+                "path": "/",
+            },
+        )
+        await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert "errors" not in result
+
+
+async def test_import_duplicate_yaml(hass: HomeAssistant) -> None:
+    """Test that the yaml import works."""
+    MockConfigEntry(
+        domain=DOMAIN,
+        data={"host": "192.168.1.123"},
+        source=config_entries.SOURCE_IMPORT,
+        unique_id="uuid",
+    ).add_to_hass(hass)
+
+    with patch(
+        "pyoctoprintapi.OctoprintClient.get_server_info",
+        return_value=True,
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_discovery_info",
+        return_value=DiscoverySettings({"upnpUuid": "uuid"}),
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.request_app_key", return_value="test-key"
+    ) as request_app_key, patch(
+        "homeassistant.components.octoprint.async_setup", return_value=True
+    ), patch(
+        "homeassistant.components.octoprint.async_setup_entry",
```

Comment:
We don't need to patch the integration setup functions on abort cases since the integration won't be set up then.

Reply:
Removed